src/swipes.py

```diff
@@ -0,0 +1,342 @@
+from datetime import date, datetime, timedelta
+from file_read_backwards import FileReadBackwards
+import json
+from os.path import isfile
+import numpy as np
+import pandas as pd
+
+from src.constants import (
+  BRB_ONLY,
+  DINING_HALL,
+  EATERY_DATA_PATH,
+  LOCATION_NAMES,
+  SCHOOL_BREAKS,
+  SWIPE_DENSITY_ROUND,
+  TABLE_COLUMNS,
+  TRILLIUM,
+  WAIT_TIME_CONVERSION,
+  WEEKDAYS,
+)
+from src.eatery import string_to_date_range
+from src.types import SwipeDataType
+
+weekdays = {v: k for k, v in WEEKDAYS.items()}  # inverting to convert strings to indexes [0,6]
+breaks = {}
+for label, dates in SCHOOL_BREAKS.items():
+  breaks[label] = string_to_date_range(dates)
+
+def parse_to_csv(file_name='data.csv'):
+  """Takes in a data.log file of swipe logs and converts them to a tabular format.
+  Creates one csv file or updates file if one already exists:
+
+  Keyword arguments:
+  file_name -- the file name to use for the output csv file (default 'data.csv')
+
+  Table columns:
+  date -- date of the swipe event, string (mm/dd/yyyy)
+  weekday -- weekday of swipe event, string (ex: 'monday', 'tuesday', etc.)
+  in_session -- whether or not school is on a break, boolean
+  location -- eatery of swipe event, string
+  start_time -- left edge of timeblock, string (hh:mm AM/PM)
+  end_time -- right edge of timeblock, string (hh:mm AM/PM)
+  swipes -- number of swipes, int
+  brb_only -- indicates if this location is a brb only eatery (used for calcs), boolean
+  dining_hall -- indicates if a location is a dining hall (used for calcs), boolean
+  trillium -- indicates if a location is Trillium (used for calcs), boolean
```

Comment:
maybe just include the multiplier as a parameter instead of having to add a new boolean for each possible one?

Reply:
not sure what you mean.  I agree that we don't want a new boolean for each one but np.multiply requires a whole boolean column to be able to operate on a condition

Reply:
i meant just including a "multipler" column and then directly multiplying it without any conditions?

Reply:
ooohhhhh shiiiiittt I get you now #bigbrainoverthere